docs/Design/open-telemetry/README.md

```diff
@@ -0,0 +1,250 @@
+adding new attributes through the OTel SDK.
+
+OTel has an extensive [collection of
+plugins](https://opentelemetry.io/ecosystem/registry/?language=js&component=instrumentation)
+to auto instrument JavaScript packages, including a number of packages that are
+used in Meadowlark including Fastify, MongoDB and PostgeSQL.
+
+## OTel Collector
+
+There are a couple of ways to handle trace data after it's been created. One way
+is to directly send telemetry data to a backend such as Prometheus, Jager or
+ElasticSearch from Meadowlark.
+
+OTel also provides the Collector. The Collector offers a vendor-agnostic
+implementation of how to receive, process and export telemetry data.
+
+Pros of using a collector:
+
+- Easy to set up (docker container available)
+- Can be run as a cluster
+- Duplicate telemetry data from the collector to multiple endpoints
+- Collects from multiple services before moving to telemetry backend
+  - Process the data (adding/removing attributes, handling batching, etc.)
+    before sending it to another destination
+- Decouple the producer with the consumer
+
+Due to the flexibility of how the collector can be set up and the control it
+gives to it's users, collectors are the suggested way to handle OTel output.
+
+## Cloud Provider Support
+
+Since the OTel Collector and popular backends are containerized, OTel can be
+used with the top three cloud providers. The following details how the big three
+integrate with OTel:
+
+### Amazon Web Services
+
+AWS has quite possibly the best built in support as of the creation of this
+document. AWS provides the 'AWS Distro' for Open Telemetry, which includes:
+
+- An AWS specific version of the OTel SDK - Allowing for the collection of AWS
+  related resources into your OTel setup
+- OTel Collector Integrates seamlessly into Amazon CloudWatch as the backend
+data store for OTel, allowing the collection of all AWS logging into a single
+location. (Cloudwatch is optional, other OTel backends can be used as well)
+
+### Microsoft Azure
+
+Microsoft plans to go all in and support OTel, but they are still building out
+the internal support for OTel. Microsoft released the Azure Monitor
+OpenTelemetry Distro at the end of May. This will allow OTel data to flow into
+Application Insights but it does require a package to be included and called
+within Meadowlark code. Without code changes it looks like a different backend
+would need to be chosen for Azure.
+
+### Google Cloud Platform
+
+GCP, like AWS has it's own collector, the OpenTelemetry Operations Collector
+that will send OTel data directly to GCP's Cloud Trace and Cloud Monitoring
+Metrics
+
+## How Meadowlark could use Open Telemetry
+
+Since we already have a form of tracing within Meadowlark through our existing
+logs, it doesn't seem like it would be all that difficult to add OTel tracing to
+Meadowlark, although managing parent/child traces may be a bit of work to do
```

Comment:
I wonder if the `traceId` that we already use in the logging would be the basis for parent/child: the initial handler that first logs the `traceId` would then be the parent and all else the child.